cli-typescript/src/utils/loaders.ts

```diff
@@ -0,0 +1,265 @@
+      console.log(chalk.green(`Signer loaded successfully: ${signer}`));
+    } catch (error) {
+      console.error(chalk.red('Error loading wallet: Check your password.'));
+      process.exit(1);
+    }
+  } else {
+    console.error(chalk.red('No password set. Skipping signer loading.'));
+    process.exit(1);
+  }
+};
+
+/**
+ * Loads default configuration values from a `defaults.json` file and sets them in `process.env`.
+ * Ensures the collections directory exists and loads the signer.
+ * @throws Error if `defaults.json` is not found.
+ */
+export const loadDefaults = async (): Promise<void> => {
+  const baseDir = process.env.BASE_DIR || path.resolve(__dirname, '..');
+  let defaultsFile = path.join(baseDir, '../defaults.json');
+
+  // Check if defaults.json exists in the expected location
+  if (!fs.existsSync(defaultsFile)) {
+    console.log(
+      chalk.yellow(
+        `defaults.json not found in ${defaultsFile}. Checking current directory...`,
+      ),
+    );
+    defaultsFile = path.resolve('./defaults.json');
+  }
+
+  if (!fs.existsSync(defaultsFile)) {
+    console.error(
+      chalk.red('defaults.json not found in current directory. Exiting...'),
+    );
+    throw new Error('defaults.json not found.');
+  }
+
+  // Read and parse defaults.json
+  const defaults = JSON.parse(fs.readFileSync(defaultsFile, 'utf-8'));
+
+  // Set default values in process.env
+  process.env.DEFAULT_COSIGNER = defaults.default_cosigner || '';
+  process.env.DEFAULT_TIMESTAMP_EXPIRY =
+    defaults.default_timestamp_expiry || '';
+  process.env.DEFAULT_MINT_CURRENCY = defaults.default_mint_currency || '';
+  process.env.DEFAULT_TOKEN_URI_SUFFIX =
+    defaults.default_token_uri_suffix || '';
+  process.env.DEFAULT_ROYALTY_RECEIVER =
+    defaults.default_royalty_receiver || '';
+  process.env.DEFAULT_ROYALTY_FEE =
+    defaults.default_royalty_fee?.toString() || '';
+  process.env.DEFAULT_MERKLE_ROOT = defaults.default_merkle_root || '';
+
+  // Create collections directory if it doesn't exist
+  const collectionsDir = path.join(baseDir, '../collections');
+  if (!fs.existsSync(collectionsDir)) {
+    console.log(
+      chalk.green(`Creating collections directory at ${collectionsDir}...`),
+    );
+    fs.mkdirSync(collectionsDir, { recursive: true });
+  }
+
+  // Load environment variables from .env file if it exists
+  const envFile = path.join(baseDir, '../.env');
+  if (fs.existsSync(envFile)) {
+    dotenv.config({ path: envFile });
+  }
+
+  // Load the signer
+  await loadSigner();
+
+  process.env.CONFIG_COMPLETE = 'true';
+};
+
+/**
+ * Loads the private key by checking for an existing keystore file or creating one interactively.
+ * @throws Error if the keystore creation fails.
+ */
+export const loadPrivateKey = async (): Promise<void> => {
+  console.log(chalk.cyan('Loading private key'));
+  const homeDir = os.homedir();
+  const magicDropKeystore =
+    process.env.MAGIC_DROP_KEYSTORE || MAGIC_DROP_KEYSTORE;
```

Comment:
We need an example env in the `cli-typescript` directory file for `MAGIC_DROP_KEYSTORE` for sure.

Reply:
Good catch. This is not supposed to be an .env, an oversight on my end. 🙂